compiler/builtins/console.ts

```diff
@@ -89,8 +89,29 @@ export const __Porffor_miniLog = (arg: any) => {
 export const __Porffor_print = (arg: any, colors: boolean = true) => {
   // todo: Symbol.toStringTag could reduce duplication here
 
-  const t: i32 = Porffor.rawType(arg);
-  switch (t) {
+  // note: this doesn't have access to the upper scope!! do not use any variables from up there
+  const __Porffor_printArray = (arg: any, colors: boolean, length: boolean = false) => {
```

Comment:
Could you put this at the top-level and exported like the others please?

Reply:
The circular dependency breaks it if I do that unfortunately

Reply:
oh oops, seems fine then :+1: